docs/how-to/use_elastic_store.md

```diff
@@ -0,0 +1,96 @@
+# How-To: Use Elastic to Store Similarity Index
+
+[ElasticStore](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-store.html]) can be used as a store in [SimilarityIndex](../concepts/similarity_indexes.md). In this guide, we will show you how to execute similarity search by using elastic search.
+In the example the elastic search engine is provided by official docker image.
+There are two approaches available to perform similarity search: Elastic Search Store and Elastic Vector Search.
+Elastic Search Store uses embeddings and kNN search to find similarities while Elastic Vector Search - semantic search - uses ELSER (Elastic Learned Sparse EncodeR) model
+to encode and search the data.
+
+## Environment setup
+
+* [Download and deploy elastic search docker image](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html). Please note that for elastic vector search elastic docker container required at least 8GB of RAM and activation
+
+```commandline
+docker network create elastic
+docker pull docker.elastic.co/elasticsearch/elasticsearch:8.13.4
+docker run --name es01 --net elastic -p 9200:9200 -it -m 8GB docker.elastic.co/elasticsearch/elasticsearch:8.13.4
+# Copy the generated elastic password and enrollment token. These credentials are only shown when you start Elasticsearch for the first time. You can regenerate the credentials using the following commands.
+docker cp es01:/usr/share/elasticsearch/config/certs/http_ca.crt .
+curl --cacert http_ca.crt -u elastic:$ELASTIC_PASSWORD https://localhost:9200
+
+# For vector search it is required to enroll to appropiate subscryption level or trial verison which supports machine learning.
+# It is required to download ELSER model for example through kibana
+docker run --name kib01 --net elastic -p 5601:5601 docker.elastic.co/kibana/kibana:8.13.4
+# [Description how to download/enable ELSER model](https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-elser.html)
+
+```
+
+* Install elastic search python client
+```commandline
+pip install dbally[elasticsearch,litllm]
+```
+
+## Implementing a SimilarityIndex
+
+To use similarity search it is required to define data fetcher and data store.
+
+### Data fetcher:
+
+```python
+class DummyCountryFetcher(SimilarityFetcher):
+    async def fetch(self):
+        return ["United States", "Canada", "Mexico"]
+```
+
+### Data store:
+Elastic store similarity search works on embeddings. For create embeddings the embedding client is passed as an argument.
+For example, we can use:
+* [OpenAiEmbeddingClient][dbally.embedding_client.OpenAiEmbeddingClient].
+
+```python
+from dbally.embeddings.litellm import LiteLLMEmbeddingClient
+
+embeddings=LiteLLMEmbeddingClient((api_key="your-api-key")
+```
+
+To implement a Similarity Index with elastic store create ElasticStore object and pass it to a store following argument:
```

Comment:
This doesn't render in the docs, please reformat this. Also, I'm not sure if we should copy paste these argument definitions from the doc string